File: Python/insertion_sort.py
import random


def insertion_sort(arr):
    length = len(arr)

    for index in range(1, length):
        key = arr[index]
        pointer = index - 1

        while key<arr[pointer] and pointer>=0:
            arr[pointer+1] = arr[pointer]
            pointer -= 1

        arr[pointer+1] = key


if __name__ == '__main__':
    random.seed(0)

    arr = [random.randint(-100,100) for _ in range(10)]
    print('Array: ', arr)

    insertion_sort(arr)
    print('Sorted array: ', arr)

File: C++/insertion_sort.cc
#include <iostream>
#include <vector>


void InsertionSort(std::vector<int>& arr){
    int length = arr.size();

    for(int index=1; index<length; index++){
        int key = arr[index];
        int pointer = index - 1;

        while(key<arr[pointer] && pointer>=0){
            arr[pointer+1] = arr[pointer];
            pointer--;
        }

        arr[pointer+1] = key;
    }
}


void Print(const std::vector<int>& arr){
    for(const auto& entry : arr){
        std::cout << entry << " ";
    }

    std::cout << std::endl;
}


int main(){
    std::vector<int> arr = {-2, 94, 7, -90, -34, 30, 24, 3, 100, -23};

    Print(arr);
    InsertionSort(arr);
    Print(arr);
}